src/module.c

```diff
@@ -3694,10 +3695,13 @@ void moduleTypeNameByID(char *name, uint64_t moduleid) {
  *   been removed from the DB by redis, and may soon be freed by a background thread. Note that 
  *   it won't be called on FLUSHALL/FLUSHDB (both sync and async), and the module can use the 
  *   RedisModuleEvent_FlushDB to hook into that.
+ * * **copy**: A callback function pointer that is used to make a copy of the specified key.
+ *   The module is expected to perform a deep copy of the specified value and return it.
+ *   In addition, hints about the names of the source and destination keys is provided.
```

Comment:
Why are hints provided? It seems like it is breaking it's not information the module should need, since keys shouldn't really know where they are. Since you can move them around after they are copied and achieve the same effect.

Reply:
we've come to realize that a lot of modules do things they shouldn't!
one case that seems semi-common is storing extra metadata on keys outside of the keyspace.
simple modules will not use that information, but the complex ones may depend on it.
p.s. these modules go through hoops to overcome the absence of the key name in the free callback, and we recently added it for the unlink callback.

Reply:
That's also why I refer to this as a hint rather than a hard commitment to provide a key name. We also have the same thing in `rdb_save` and `rdb_load` where the key name can be queried from `RedisModuleIO` but it may or may not be there depending on the context.

Reply:
Yeah, the whole notion that we added a "hint" is sort of weird. I don't know why anyone would actually want to rely on it. If they want to store additional metadata outside of the keyspace, it seems like the notification system is a better approach?